airflow-core/src/airflow/ui/src/pages/XCom/XComEntry.tsx

```diff
@@ -86,16 +87,16 @@ export const XComEntry = ({ dagId, mapIndex, runId, taskId, xcomKey }: XComEntry
     />
   ) : (
     <HStack>
-      {["array", "object"].includes(typeof data?.value) ? (
-        <RenderedJsonField content={data?.value as object} enableClipboard={false} />
+      {isObjectOrArray ? (
+        <RenderedJsonField content={xcomValue as object} enableClipboard={false} />
       ) : (
         <Text>{renderTextWithLinks(valueFormatted)}</Text>
       )}
-      {Boolean(data?.value) ? (
+      {xcomValue === undefined ? undefined : (
```

Comment:
Nit: `null` does nto make sense to copy, can you remove the clipboard icon when `null` value?

Reply:
Done. 
<img width="580" height="178" alt="image" src="https://github.com/user-attachments/assets/4aabc6f4-967f-4dc6-8e26-0ab34c2f634a" />